FreeRTOS_Sockets.c

```diff
@@ -1599,6 +1599,25 @@ void * vSocketClose( FreeRTOS_Socket_t * pxSocket )
         FreeRTOS_Socket_t * pxOtherSocket;
         uint16_t usLocalPort = pxSocketToDelete->usLocalPort;
 
+        if( pxSocketToDelete->u.xTCP.ucTCPState == ( uint8_t ) eTCP_LISTEN )
+        {
+            for( pxIterator = listGET_NEXT( pxEnd );
+                 pxIterator != pxEnd; )
+            {
+                pxOtherSocket = ipCAST_PTR_TO_TYPE_PTR( FreeRTOS_Socket_t, listGET_LIST_ITEM_OWNER( pxIterator ) );
+                pxIterator = listGET_NEXT( pxIterator );
+
+                if( ( pxOtherSocket->u.xTCP.ucTCPState != ( uint8_t ) eTCP_LISTEN ) &&
+                    ( pxOtherSocket->usLocalPort == usLocalPort ) &&
+                    ( pxOtherSocket->u.xTCP.bits.bPassQueued || pxOtherSocket->u.xTCP.bits.bPassAccept ) )
+                {
+                    vSocketClose( pxOtherSocket );
+                }
+            }
+
+            return;
```

Comment:
MISRA doesn't like a `return` in the middle of a function, so could you change it into an else?